core/src/main/scala/kafka/server/BrokerLifecycleManager.scala

```diff
@@ -47,7 +47,7 @@ import scala.concurrent.duration._
  */
 trait BrokerLifecycleManager {
 
-  // Initiate broker registration and start the heartbeat scheduler loop
+  // Start the heartbeat scheduler loop with broker registration as initial state
```

Comment:
broker registration is not the initial state, though.  as per the KIP, we start off in NOT_RUNNING